### PR TITLE
Moved EvalSum, Rotation, Bootstrap and Automorphism keys to the same map

### PR DESCRIPTION
**Pull request by dsuponitskiy**

### CODE REVIEW

**Review by pascoec — APPROVED** [review]

File: src/pke/include/cryptocontext.h [hunk 1]
```diff
@@ -2701,6 +2689,17 @@ class CryptoContextImpl : public Serializable {
     std::shared_ptr<std::map<usint, EvalKey<Element>>> EvalSumRowsKeyGen(const PrivateKey<Element> privateKey,
                                                                          const PublicKey<Element> publicKey = nullptr,
                                                                          usint rowSize = 0, usint subringDim = 0);
+
+    // [[deprecated(
```
Comment (yspolyakov) [inline src/pke/include/cryptocontext.h:2693]: Why is the deprecation logic commented out here and other places?

Comment (dsuponitskiy) [reply src/pke/include/cryptocontext.h:2693]: It is the new code waiting to be un-commented after publicKey is removed. I can remove it

**Comment by yspolyakov** [comment]
We should add the same logic for `MultiEvalAutomorphismKeyGen` and multiparty key gen methods based on it. Should we do it as part of a separate issue/PR?

File: src/pke/include/cryptocontext.h (hunk 1 above)
Comment (yspolyakov) [reply src/pke/include/cryptocontext.h:2693]: This is fine. It is OK to keep it as is.

**Review by yspolyakov — APPROVED** [review]

**Pull request merged**